src/apolloSentryLink.ts

```diff
@@ -0,0 +1,89 @@
+import { ApolloLink } from '@apollo/client/core';
+import { Severity } from '@sentry/types';
+import deepMerge from 'deepmerge';
+
+import { makeOperation } from './operation';
+import { Breadcrumb, makeBreadcrumb } from './breadcrumb';
+import { defaultOptions, ApolloLinkSentryOptions, Options } from './options';
+import {
+  attachBreadcrumbToSentry,
+  setFingerprint,
+  setTransaction,
+} from './sentry';
+
+export function makeSentryLink(options: Options = {}): ApolloLink {
+  return new ApolloLink((apolloOperation, forward) => {
+    const fullOptions = deepMerge(
+      defaultOptions,
+      options,
+    ) as ApolloLinkSentryOptions;
+
+    const operation = makeOperation(apolloOperation);
+
+    if (typeof fullOptions.shouldHandleOperation === 'function') {
+      const shouldIncludeOperation = fullOptions.shouldHandleOperation(
+        operation,
+      );
+      if (!shouldIncludeOperation) return forward(apolloOperation);
+    }
+
+    if (fullOptions.setTransaction) {
+      setTransaction(operation);
+    }
+
+    if (fullOptions.setFingerprint) {
+      setFingerprint(operation);
+    }
+
+    const breadcrumb = fullOptions.attachBreadcrumbs
+      ? makeBreadcrumb(operation, fullOptions.attachBreadcrumbs)
+      : undefined;
+
+    const observable = forward(apolloOperation);
+
+    observable.subscribe({
+      next: (result) => {
+        if (
+          fullOptions.attachBreadcrumbs &&
+          fullOptions.attachBreadcrumbs.includeResult
+        ) {
+          // We must have a breadcrumb if attachBreadcrumbs was set
+          (breadcrumb as Breadcrumb).data.result = result;
```

Comment:
Perhaps this can be done in a more elegant way. Orginally, I had a check for `&& breadcrumb` above, but it is actually unnecessary. This version has no cost at runtime.

Reply:
Agree it's not the most beautiful way, but it's also not a huge issue for me.